python/cudf/cudf/core/column/column.py

```diff
@@ -1554,6 +1554,17 @@ def build_column(
             offset=offset,
             null_count=null_count,
         )
+    elif is_datetime64tz_dtype(dtype):
+        if data is None:
+            raise TypeError("Must specify data buffer")
```

Comment:
`ValueError` seems more natural here unless you're matching something in pandas.
```suggestion
            raise ValueError("Must specify data buffer")
```